pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -594,21 +604,14 @@ func (nc *NodeController) monitorNodeStatus() error {
 			// Check eviction timeout.
 			if lastReadyCondition.Status == api.ConditionFalse &&
 				nc.now().After(nc.nodeStatusMap[node.Name].readyTransitionTimestamp.Add(nc.podEvictionTimeout)) {
-				// Node stays in not ready for at least 'podEvictionTimeout' - evict all pods on the unhealthy node.
-				// Makes sure we are not removing pods from to many nodes in the same time.
-				glog.Infof("Evicting pods: %v is later than %v + %v", nc.now(), nc.nodeStatusMap[node.Name].readyTransitionTimestamp, nc.podEvictionTimeout)
-				if nc.deletingPodsRateLimiter.CanAccept() {
-					nc.deletePods(node.Name)
-				}
+				nc.podEvictor.AddNodeToEvict(node.Name)
 			}
 			if lastReadyCondition.Status == api.ConditionUnknown &&
 				nc.now().After(nc.nodeStatusMap[node.Name].probeTimestamp.Add(nc.podEvictionTimeout-gracePeriod)) {
-				// Same as above. Note however, since condition unknown is posted by node controller, which means we
-				// need to substract monitoring grace period in order to get the real 'podEvictionTimeout'.
-				glog.Infof("Evicting pods2: %v is later than %v + %v", nc.now(), nc.nodeStatusMap[node.Name].readyTransitionTimestamp, nc.podEvictionTimeout-gracePeriod)
-				if nc.deletingPodsRateLimiter.CanAccept() {
-					nc.deletePods(node.Name)
-				}
+				nc.podEvictor.AddNodeToEvict(node.Name)
```

Comment:
How often does this happen?

As nodes go down, the nodecontroller is going to start periodically listing all pods once per bad node. This is a pretty heavy hammer and it would be nice to avoid it. Don't mean to overengineer this pr, but do you have any idea of what effect this has on the cluster? i.e how long does a single deletePods take as nodes go from 100->0 (its probably gated on list performance, which is representative of list performance of everything across the cluster).